src/index.js

```diff
@@ -1,203 +1,83 @@
+  );
 
-    commitTextUpdate(textInstance: Text, oldText: string, newText: string) {
-      textInstance.nodeValue = newText;
-    },
+  const root = ContainerMap.get(domContainer);
 
-    resetTextContent(domElement: Element): void {
-      domElement.textContent = '';
-    },
+  if (!root) return false;
 
-    appendChild(parentInstance: Element, child: Element | Text): void {
-      parentInstance.appendChild(child);
-    },
+  DOMLiteReconciler.unbatchedUpdates(() => {
+    root.unmount(() => {
+      ContainerMap.delete(domContainer);
+    });
+  });
 
-    appendChildToContainer(
-      parentInstance: DOMContainer,
-      child: Element | Text
-    ): void {
-      parentInstance.appendChild(child);
-    },
-    insertBefore(
-      parentInstance: Element,
-      child: Element | Text,
-      beforeChild: Element | Text
-    ): void {
-      parentInstance.insertBefore(child, beforeChild);
-    },
+  return true;
+}
 
-    insertInContainerBefore(
-      container: Element,
-      child: Element | Text,
-      beforeChild: Element | Text
-    ): void {
-      container.insertBefore(child, beforeChild);
-    },
+function findDOMNode(
+  componentOrElement: Element | ?React$Component<any, any>
+): null | Element | Text {
+  if (componentOrElement == null) return null;
 
-    removeChild(parentInstance: Element, child: Element | Text): void {
-      parentInstance.removeChild(child);
-    },
-    removeChildFromContainer(
-      parentInstance: DOMContainer,
-      child: Element | Text
-    ): void {
-      parentInstance.removeChild(child);
-    }
+  if (componentOrElement.nodeType === 1 || componentOrElement === 3) {
+    return (componentOrElement: any);
   }
-};
-
-const DOMLiteRenderer = Reconciler(hostConfig);
-
-DOMLiteRenderer.injectIntoDevTools({
-  bundleType: 1,
-  version: '0.1.0',
-  rendererPackageName: 'react-dom-lite',
-  findFiberByHostInstance: getInternalHandleFromInstance
-});
-
-let ContainerMap = new WeakMap();
-function render(elements: React$Element<any>, domContainer: Element) {
-  const container = DOMLiteRenderer.createContainer(domContainer, false, false);
-  const root = new Root(container, DOMLiteRenderer);
-
-  ContainerMap.set(domContainer, root);
+  // $FlowFixMe
+  const handler: OpaqueHandle = componentOrElement._reactInternalFiber;
+  if (handler) return DOMLiteReconciler.findHostInstance(handler);
+
+  invariant(
+    false,
+    'Element appears to be neither ReactComponent nor DOMNode. Keys: %s',
+    Object.keys(componentOrElement)
+  );
+}
 
-  root.render(elements);
+// FIXME: Upstream needs to provide a better API for this.
+function createPortal(
+  children: ReactNodeList,
+  container: DOMContainer,
+  key?: string
+): ReactPortal {
+  return {
+    $$typeof:
+      typeof Symbol === 'function' && Symbol.for
+        ? Symbol.for('react.portal')
+        : 0xeaca,
+    key: key == null ? null : String(key),
+    children,
+    containerInfo: container,
+    implementation: null
+  };
```

Comment:
ditto here. I'm imagining `implementation` is meant to solve this, but the core method isn't exposed publiclly anywhere. Probably should go in react-reconciler?

Reply:
No, you can ignore this field. It was meant for cross-renderer portals that didn't get implemented yet. For now this is unused.

Reply:
>There isn't anything specific tho to the DOM support of portals (a part from providing the various methods in the reconciler config)?

I think it should just work.

>One interesting thing here is tho is that event propagation is not gonna work since we aren't using the event system.

Yeah.